pkg/azureutils/azure_disk_utils.go

```diff
@@ -1090,68 +1090,29 @@ func GetAzVolumeAttachmentState(volumeAttachmentStatus storagev1.VolumeAttachmen
 	var err error
 
 	curNetRetry := 0
 	curRetry := 0
 
-	objName := obj.GetName()
 	ctx, w := workflow.New(ctx, workflow.WithCaller(1))
 	defer func() {
 		w.AddDetailToLogger(consts.RetryKey, curRetry, consts.NetRetryKey, curNetRetry)
 		w.Finish(err)
 	}()
 
+	needGetOriginalObj := false
 	var updatedObj client.Object
 
 	conditionFunc := func() error {
 		var err error
-		var originalObj client.Object
-		switch target := obj.(type) {
-		case *azdiskv1beta2.AzVolume:
-			if informerFactory != nil {
-				originalObj, err = informerFactory.Disk().V1beta2().AzVolumes().Lister().AzVolumes(target.Namespace).Get(objName)
-			} else if cachedClient != nil {
-				originalObj = &azdiskv1beta2.AzVolume{}
-				err = cachedClient.Get(ctx, types.NamespacedName{Namespace: target.Namespace, Name: objName}, originalObj)
-			} else {
-				originalObj, err = azDiskClient.DiskV1beta2().AzVolumes(target.Namespace).Get(ctx, objName, metav1.GetOptions{})
-			}
-		case *azdiskv1beta2.AzVolumeAttachment:
-			if informerFactory != nil {
-				originalObj, err = informerFactory.Disk().V1beta2().AzVolumeAttachments().Lister().AzVolumeAttachments(target.Namespace).Get(objName)
-			} else if cachedClient != nil {
-				originalObj = &azdiskv1beta2.AzVolumeAttachment{}
-				err = cachedClient.Get(ctx, types.NamespacedName{Namespace: target.Namespace, Name: objName}, originalObj)
-			} else {
-				originalObj, err = azDiskClient.DiskV1beta2().AzVolumeAttachments(target.Namespace).Get(ctx, objName, metav1.GetOptions{})
-			}
-		case *azdiskv1beta2.AzDriverNode:
-			if informerFactory != nil {
-				originalObj, err = informerFactory.Disk().V1beta2().AzDriverNodes().Lister().AzDriverNodes(target.Namespace).Get(objName)
-			} else if cachedClient != nil {
-				originalObj = &azdiskv1beta2.AzDriverNode{}
-				err = cachedClient.Get(ctx, types.NamespacedName{Namespace: target.Namespace, Name: objName}, originalObj)
-			} else {
-				originalObj, err = azDiskClient.DiskV1beta2().AzDriverNodes(target.Namespace).Get(ctx, objName, metav1.GetOptions{})
-			}
-		case *storagev1.VolumeAttachment:
-			if cachedClient != nil {
-				originalObj = &storagev1.VolumeAttachment{}
-				err = cachedClient.Get(ctx, types.NamespacedName{Namespace: target.Namespace, Name: objName}, originalObj)
-			} else {
-				return status.Errorf(codes.Internal, "cannot update VolumeAttachment object if controller runtime client is not provided.")
-			}
-		default:
-			return status.Errorf(codes.Internal, "object (%v) not supported.", reflect.TypeOf(target))
-		}
 
-		if err != nil {
-			if !k8serrors.IsNotFound(err) {
-				err = status.Errorf(codes.Internal, "failed to get object: %v", err)
+		if originalObj == nil || needGetOriginalObj {
```

Comment:
Instead of maintaining a separate Boolean value, you can infer the need to get the original object from `curRetry`, i.e. `curRetry > 0`. This is because `curRetry` is updated on conflict errors in the `isRetriable` function below (line 1190). This would avoid the need to check the error explicitly in `conditionFunc`.